fuse_examples/segmentation/runner_seg.py

```diff
@@ -0,0 +1,488 @@
+from fuse.utils.utils_param_sampler import FuseUtilsParamSamplerUniform as Uniform
+from fuse.utils.utils_param_sampler import FuseUtilsParamSamplerRandBool as RandBool
+from fuse.utils.utils_param_sampler import FuseUtilsParamSamplerRandInt as RandInt
+from fuse.utils.utils_gpu import FuseUtilsGPU
+from fuse.utils.utils_logger import fuse_logger_start
+from fuse.data.augmentor.augmentor_default import FuseAugmentorDefault
+from fuse.data.visualizer.visualizer_default import FuseVisualizerDefault
+from fuse.data.dataset.dataset_default import FuseDatasetDefault
+from fuse.models.model_wrapper import FuseModelWrapper
+from fuse.losses.segmentation.loss_dice import DiceBCELoss
+from fuse.losses.segmentation.loss_dice import FuseDiceLoss
+from fuse.losses.loss_default import FuseLossDefault
+from fuse.managers.manager_default import FuseManagerDefault
+from fuse.managers.callbacks.callback_tensorboard import FuseTensorboardCallback
+from fuse.managers.callbacks.callback_metric_statistics import FuseMetricStatisticsCallback
+from fuse.managers.callbacks.callback_time_statistics import FuseTimeStatisticsCallback
+from fuse.data.processor.processor_dataframe import FuseProcessorDataFrame
+# from fuse.metrics.metric_auc_per_pixel import FuseMetricAUCPerPixel
+# from fuse.metrics.segmentation.metric_score_map import FuseMetricScoreMap
+# from fuse.analyzer.analyzer_default import FuseAnalyzerDefault
+from fuse.eval.evaluator import EvaluatorDefault
+from fuse.eval.metrics.segmentation.metrics_segmentation_common import MetricDice, MetricIouJaccard, MetricOverlap, Metric2DHausdorff, MetricPixelAccuracy
+
+from data_source_segmentation import FuseDataSourceSeg
+from seg_input_processor import SegInputProcessor
+
+from unet import UNet
+
+
+def perform_softmax(output):
+    if isinstance(output, torch.Tensor):  # validation
+        logits = output
+    else:  # train
+        logits = output.logits
+    cls_preds = F.softmax(logits, dim=1)
+    return logits, cls_preds
+
+
+SZ = 512
+TRAIN = f'../../data/siim/data{SZ}/train/'
+TEST = f'../../data/siim/data{SZ}/test/'
+MASKS = f'../../data/siim/data{SZ}/masks/'
+
+# TODO: Path to save model
+ROOT = '../results/'
+# TODO: path to store the data  
+ROOT_DATA = ROOT
+# TODO: Name of the experiment
+EXPERIMENT = 'unet_seg_results'
+# TODO: Path to cache data
+CACHE_PATH = '../results/'
+# TODO: Name of the cached data folder
+EXPERIMENT_CACHE = 'exp_cache'
+
+PATHS = {'data_dir': [TRAIN, MASKS, TEST],
+         'model_dir': os.path.join(ROOT, EXPERIMENT, 'model_dir'),
+         'force_reset_model_dir': True,  # If True will reset model dir automatically - otherwise will prompt 'are you sure' message.
+         'cache_dir': os.path.join(CACHE_PATH, EXPERIMENT_CACHE+'_cache_dir'),
+         'inference_dir': os.path.join(ROOT, EXPERIMENT, 'infer_dir'),
+         'analyze_dir': os.path.join(ROOT, EXPERIMENT, 'analyze_dir')}
```

Comment:
We use the evaluator now, Let's change terminology,
Please follow other examples.